tests/apollo/test_skvbc_reconfiguration.py

```diff
@@ -555,6 +555,228 @@ async def test_addRemove_command(self, bft_network):
+        # All next request should be go through the slow path
+        for i in range(100):
+            await skvbc.write_known_kv()
+        key, val = await skvbc.write_known_kv()
+        live_replicas = bft_network.all_replicas(without=crashed_replicas)
+        client = bft_network.random_client()
+        client.config._replace(req_timeout_milli=10000)
+        checkpoint_before = await bft_network.wait_for_checkpoint(replica_id=0)
+        op = operator.Operator(bft_network.config, client,  bft_network.builddir)
+        test_config = 'new_configuration_n_4_f_1_c_0'
+        await op.add_remove_with_wedge(test_config)
+        await self.verify_replicas_are_in_wedged_checkpoint(bft_network, checkpoint_before, range(len(live_replicas)))
+        expectedSeqNum = (checkpoint_before  + 2) * 150
+        for r in bft_network.all_replicas(without=crashed_replicas):
+            lastExecSn = await bft_network.get_metric(r, bft_network, "Gauges", "lastExecutedSeqNum")
+            self.assertEqual(expectedSeqNum, lastExecSn)
+        bft_network.stop_all_replicas()
+        # We now expect the replicas to start with a fresh new configuration which means that we
+        # need to see in the logs that isNewStorage() = true. Also,
+        # we expect to see that lastStableSeqNum = 0 (for example)
+        conf = TestConfig(n=4,
+                f=1,
+                c=0,
+                num_clients=10,
+                key_file_prefix=KEY_FILE_PREFIX,
+                start_replica_cmd=start_replica_cmd,
+                stop_replica_cmd=None,
+                num_ro_replicas=0)
+        await bft_network.change_configuration(conf)
+        bft_network.start_all_replicas()
+        for r in bft_network.all_replicas():
+            last_stable_checkpoint = await bft_network.get_metric(r, bft_network, "Gauges", "lastStableSeqNum")
+            self.assertEqual(last_stable_checkpoint, 0)
+        await self.validate_state_consistency(skvbc, key, val)
+        for i in range(100):
+            await skvbc.write_known_kv()
+        for r in bft_network.all_replicas():
+            assert (r < 4)
+            nb_fast_path = await bft_network.get_metric(r, bft_network, "Counters", "totalFastPaths")
+            self.assertGreater(nb_fast_path, 0)
+    
+    @with_trio
+    @with_bft_network(start_replica_cmd, selected_configs=lambda n, f, c: n == 7)
+    async def test_add_nodes(self, bft_network):
+        """
+             Sends a addRemove command and checks that new configuration is written to blockchain.
+             Note that in this test we assume no failures and synchronized network.
+             The test does the following:
+             1. A client sends a add node command which will also wedge the system on next next checkpoint
+             2. Validate that all replicas have stopped
+             3. Load a new configuration to the bft network
+             4. Add node is done in phases, (n=4,f=1,c=0)->(n=6,f=1,c=0)->(n=7,f=2,c=0)
```

Comment:
I would like to have another test (maybe in a different PR) in which we use the state of one replica for the new 3 replicas such that we move directly from 4->7 replicas.
This can be done by copying the rocksdb data to the directories of the new replicas.
Once the new replicas are starting they will remove the old metadata and they should start correctly.
Note that this may require changes in the bft module